Userland/Libraries/LibGfx/ImageFormats/PBMLoader.cpp

```diff
@@ -12,19 +12,19 @@ namespace Gfx {
 
 ErrorOr<void> read_image_data(PBMLoadingContext& context)
 {
+    TRY(create_bitmap(context));
+
     auto& stream = *context.stream;
-    Vector<Gfx::Color> color_data;
 
     auto const context_size = context.width * context.height;
-    color_data.resize(context_size);
 
     if (context.type == PBMLoadingContext::Type::ASCII) {
         for (u64 i = 0; i < context_size; ++i) {
             auto const byte = TRY(stream.read_value<u8>());
             if (byte == '0')
-                color_data[i] = Color::White;
+                context.bitmap->set_pixel(i % context.width, i / context.width, Color::White);
```

Comment:
You could also do `bitmap->begin()[i] = Color::White.value()` (doesn't bounds check then though)

Reply:
I'm not a fan as it also doesn't check for the storage type. I know that it shouldn't be an issue here, but let's keep it the safe way until we have something better